### ISSUE TITLE
[BUG] dot notation not interpreted by a trigger when statement.

### ISSUE THREAD

**Issue by martijnburger** [formio/formio#998]
## Environment
Please provide as many details as you can:

- Hosting type
  - [x] Form.io
  - [ ] Local deployment
    - Version:
- Formio.js version: 3.28.1
- Frontend framework: vuejs
- Browser: Chrome
- Browser version: 81.0.4044.138

## Steps to Reproduce
1. Create a form with two components:
```
{
	"type": "column",
	"width": 6,
	"components": [
		{
			"key": "requestedCovers.HOUSECONTENT_JEWELRY",
			"label": "Lijfsieraden",
			"type": "checkbox"
		}
	]
},
{
	"type": "column",
	"width": 6,
	"components": [
		{
			"key": "insuredAmounts.HOUSECONTENT_JEWELRY",
			"label": "Lijfsieraden",
			"hideLabel": true,
			"type": "currency",
			"currency": "EUR",
			"defaultValue": 0,
			"logic": [
				{
					"trigger": {
						"type": "simple",
						"simple": {
							"show": true,
							"when": "requestedCovers.HOUSECONTENT_JEWELRY",
							"eq": "false"
						}
					},
					"actions": [
						{
							"type": "property",
							"property": {
								"value": "disabled",
								"type": "boolean"
							},
							"state": "true"
						}
					]
				}
			]
		}
	]
},
```
2. Open the created form.
3. Select and deselect the checkbox

## Expected behavior
When the checkbox is deselected the currency field is disabled and when the checkbox is selected the currency field is enabled.

## Observed behavior
The currency field is always enabled. To workaround this problem:
(1) Rename the checkbox key to something that does not include the dot notation (i.e. requestedCovers-HOUSECONTENT_JEWELRY. No option in our situation because the backend api cannot be changed to this.
(2) Remove the "requestedCovers." from the when statement. No option in our situation because that gives a name collision with the `"key": "insuredAmounts.HOUSECONTENT_JEWELRY"`

**Comment by jhen1422** [comment]
@martijnburger Thank you for the report. We're able to replicate on our end and have added a ticket to our backlog. Please note, we welcome pull requests in this library. If you need this item expedited, please feel free to contact support@form.io for pricing.

**Comment by jhen1422** [comment]
@martijnburger 
This issue has been resolved in version 4.10.0-rc.9
